unitylibs/core/workflow/workflow-firefly/widget.js

```diff
@@ -540,4 +624,293 @@ export default class UnityWidget {
+          const useBtn = prevRow?.querySelector('.use-prompt-btn.inline');
+          if (useBtn) this.consumeEventAndFocus(ev, useBtn);
+        }
+      }
+    });
+
+    tile.append(player);
+    return tile;
+  }
+
+  resetAllSoundVariations(rootEl) {
+    const root = rootEl || this.widget;
+    root.querySelectorAll('.variation-tile').forEach((t) => { this.resetTileToIdle(t); });
+    root.querySelectorAll('.sound-details').forEach((d) => d.remove());
+    root.querySelectorAll('.drop-item.sound-expanded').forEach((el) => el.classList.remove('sound-expanded'));
+    root.querySelectorAll('.drop-item .use-prompt-btn.inline').forEach((b) => b.remove());
+  }
+
+  toggleSoundDetails(dropdown, item, promptObj, promptIndex) {
+    const next = item.nextElementSibling;
+    if (next && next.classList.contains('sound-details')) {
+      next.querySelectorAll('.variation-tile').forEach((t) => { this.resetTileToIdle(t); });
+      next.remove();
+      item.classList.remove('sound-expanded');
+      const inlineExisting = item.querySelector('.use-prompt-btn.inline');
+      if (inlineExisting) inlineExisting.remove();
+      return;
+    }
+    this.resetAllSoundVariations(dropdown);
+    const inlineBtn = createTag('button', {
+      class: 'use-prompt-btn inline',
+      'data-prompt-index': String(promptIndex),
+      'aria-label': `Use prompt ${promptIndex}: ${promptObj.prompt}`,
+    }, 'Use prompt');
+    inlineBtn.addEventListener('click', (e) => {
+      e.preventDefault();
+      e.stopPropagation();
+      this.hidePromptDropdown();
+      const btn = this.genBtn;
+      if (btn) {
+        btn.dataset.soundPrompt = promptObj.prompt;
+        btn.click();
+      }
+    });
+    inlineBtn.addEventListener('keydown', (e) => {
+      if (e.key === 'Enter' || e.key === ' ') {
+        e.preventDefault();
+        e.stopPropagation();
+        inlineBtn.click();
+        return;
+      }
+      if (e.key === 'Tab' && !e.shiftKey) {
+        const detailsEl = item.nextElementSibling && item.nextElementSibling.classList?.contains('sound-details')
+          ? item.nextElementSibling
+          : null;
+        const firstTile = detailsEl?.querySelector('.variation-tile');
+        if (firstTile) { this.consumeEventAndFocus(e, firstTile); }
+      }
+      if (e.key === 'Tab' && e.shiftKey) { this.consumeEventAndFocus(e, item); }
+    });
+    item.classList.add('sound-expanded');
+    item.append(inlineBtn);
+    const details = this.renderSoundDetails(promptObj);
+    item.after(details);
+  }
+
+  renderSoundDetails(promptObj) {
+    const details = createTag('div', { class: 'sound-details', role: 'region' });
+    const strip = createTag('div', { class: 'variation-strip' });
+    const vars = Array.isArray(promptObj.variations) ? promptObj.variations : [];
+    vars.forEach((v, i) => { const tile = this.createVariationTile(v, i, strip); strip.append(tile); });
+    details.append(strip);
+    return details;
+  }
```

Comment:
Does it make sense to have all of this as a helper function that lives outside of widget.js and is only loaded if sound is being used? 

If we are worried about a few extra lines for lana and performance this is a lot of extra lines for something that is only used on a few pages.